src/lib/InstantSearch.js

```diff
@@ -340,6 +340,7 @@ class InstantSearch extends EventEmitter {
   dispose() {
     this.removeWidgets(this.widgets);
     this.started = false;
+    this.helper = null;
```

Comment:
Can we add a comment explaining why this line is necessary? (so that the new search is performed with a new helper state).

It'd be nice to add some context there for future contributors.